core/object/object.cpp

```diff
@@ -254,10 +254,44 @@ Object::Connection::Connection(const Variant &p_variant) {
 bool Object::_predelete() {
 	_predelete_ok = 1;
 	notification(NOTIFICATION_PREDELETE, true);
-	if (_predelete_ok) {
-		_class_name_ptr = nullptr; // Must restore, so constructors/destructors have proper class name access at each stage.
-		notification(NOTIFICATION_PREDELETE_CLEANUP, true);
+	if (!_predelete_ok) {
+		return false;
+	}
+
+	_class_name_ptr = nullptr; // Must restore, so constructors/destructors have proper class name access at each stage.
+	notification(NOTIFICATION_PREDELETE_CLEANUP, true);
+
+	// Deletion order objects start with the most derived class, and progresses towards the base Object class:
```

Comment:
This one's on me but
```suggestion
	// Destruction order objects start with the most derived class, and progresses towards the base Object class:
```